test/testsuites/csctargetnamespacetests.go

```diff
@@ -0,0 +1,119 @@
+package testsuites
+
+import (
+	"fmt"
+	"testing"
+
+	operator "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
+	"github.com/operator-framework/operator-marketplace/test/helpers"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// CscTargetNamespace is a test suit that confirms that the targetNamespace field within
+// a CatalogSourceConfig is handled correctly.
+func CscTargetNamespace(t *testing.T) {
+	t.Run("non-existing-target-namespace", testCscWithNonExistingTargetNamespace)
+}
+
+func testCscWithNonExistingTargetNamespace(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	// Get global framework variables.
+	f := test.Global
+
+	// Run tests.
+	if err := cscWithNonExistingTargetNamespace(t, f, ctx); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func cscWithNonExistingTargetNamespace(t *testing.T, f *test.Framework, ctx *test.TestCtx) error {
+	// Get test namespace.
+	namespace, err := test.NewTestCtx(t).GetNamespace()
```

Comment:
Why are you testing for this in all your tests? Can't it be done in just one place?

Reply:
the `t.run` function doesn't allow you to pass namespaces into the call and a namespace is required when using the client. We could create a global variable in the `helpers package` and provide a method that allows you to set/get the namespace.

Reply:
Might not be worth it. Please wait for feedback from others.

Reply:
I'd prefer to leave it as is if we can't pass it around. I don't think it's a guarantee that every test will use the same namespace.

Reply:
@kevinrizza this is a guarantee right now because we use the [--namespace](https://github.com/operator-framework/operator-marketplace/blob/master/scripts/e2e-tests.sh#L8) tag which [forces a all tests to be ran in a single namespace](https://github.com/operator-framework/operator-sdk/blob/master/doc/test-framework/writing-e2e-tests.md#namespace-flag).

Reply:
This is only a guarantee that the operator is running in the namespace. I suppose from the test context call that is true, but there's nothing stopping you from writing code that touches multiple namespaces.

Reply:
Yep that's true - anyways my vote is to leave as is for now since the benefit seems minimal.